mode-apis-base/src/alertingModels.ts

```diff
@@ -163,12 +163,47 @@ export const isMetricsThresholdAlertCondition = (obj: unknown): obj is MetricsTh
 
 
 
-/**
- * TODO - Define the attributes required for AlertAction
- */
-export interface AlertAction {
+export interface AlertMessageTemplate {
+    readonly title: string;
+    readonly message: string;
+    readonly recoveryMessage?: string;
+}
+
+export type AlertRecipientType = 'email';
+
+export interface AlertRecipient {
+    recipientType: AlertRecipientType;
 }
 
+export interface AlertRecipientEmail extends AlertRecipient {
+    readonly recipientType: 'email';
+    readonly email: string;
+}
+
+// We may want to have more various recipient types in the future.
+//
+// export type AlertRecipientType = 'email' | 'agent' | 'event';
+//
+// export interface AlertRecipientAgent extends AlertRecipient {
+//     readonly recipientType: 'agent';
+//     readonly agentId: number;
+// }
+//
+// export interface AlertRecipientEvent extends AlertRecipient {
+//     readonly recipientType: 'event';
+//     readonly eventType: string;
+// }
+
+export interface AlertFilter {
+    // TODO: implement this later
+    // AlertFilter let you filter alerts by some criteria.
+    // For example, you can filter alerts by severity, or by alert condition type, or entity types.
+}
+
+export interface AlertAction {
+    readonly filter?: AlertFilter;
+    readonly recipients: readonly [AlertRecipient, ...AlertRecipient[]];
+}
```

Comment:
Just to confirm how we are planning to use the filter. I am guessing we plan to do something like this?

```
alertRule = {
    name: "...",
    condition: ...,

    actions: [
        // If the alert is LOW severity, only need to send a message to Slack
        {
            filter: {
                severity: "low",
            },
            recipients: [{
                recipientType: "slack",
                channel: "devops"
            }]
        },

        // If the alert is HIGH severity, send a message to Slack AND email ops
        {
            filter: {
                severity: "high",
            },
            recipients: [{
                recipientType: "email",
                email: "ops@tinkermode.com"
            }, {
                recipientType: "slack",
                channel: "devops"
            }
        ]
    }];
}
```

Reply:
Exactly! The other example filter is by the specific entity.

```
filter: {
  entity: "tyo_office"
}
```